pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -888,6 +958,8 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPListenerOptsForPor
 			return false, nil
 		}
 		if pluginParams.Service != nil {
+			// If the service port was defined as unknown. It will conflict with all other
+			// protocols.
```

Comment:
isnt this one place where sniffing is useful?

Reply:
If the protocol of current listener entry is TCP and we are building HTTP filter chains on the same port, we should use protocol sniffing rather than reporting conflict. Right?

Reply:
Make some change to enable conflict.